http3/client.go

```diff
@@ -273,3 +273,24 @@ func (c *client) doRequest(
 
 	return res, requestError{}
 }
+
+// Take an http client and replace the transport with an http3 roundtripper.
+// Clones the TLS config from the client if it exists or else uses the default.
+func UpgradeClient(client *http.Client) error {
+	transport, ok := client.Transport.(*http.Transport)
+	if !ok {
+		return errors.New("could not extract transport from client, must be of type *http.Transport")
+	}
+	tlsConf := transport.TLSClientConfig
+	if tlsConf == nil {
+		tlsConf = &tls.Config{}
+	} else {
+		tlsConf = tlsConf.Clone()
+	}
```

Comment:
Why is it necessary to clone the `tls.Config` here?

Reply:
It was requested in the initial request, https://github.com/tsenart/vegeta/pull/453/files#r339342957 

I am happy to remove it as I don't fully understand the nuance involved in copying the TLS config. Is it acceptable to use a fresh one every time.

Reply:
We already clone the config here, before we modify it: https://github.com/lucas-clemente/quic-go/blob/b034f1ac7c9b66ad5814e747d43d7c7915310ff3/http3/client.go#L59-L70

I don't think we need another `Clone` here, unless you want to modify fields.